questions.ts

```diff
@@ -625,122 +641,210 @@ export const askQuestions = async (
+                return `URL=${url} has invalid protocol, only allow http or https`;
+              }
+            }
+            return true;
+          };
+
+          const dataSource: TemplateDataSource = {
+            type: "web",
+            config: [] as WebSourceConfig[],
+          };
+
+          while (true) {
+            const questions: any[] = [
+              {
+                type: "text",
+                name: "baseUrl",
+                message: "Please provide base URL of the website: ",
+                initial: "https://www.llamaindex.ai",
+                validate: (value: string) => {
+                  if (!value.includes("://")) {
+                    value = `https://${value}`;
+                  }
+                  const urlObj = new URL(value);
+                  if (
+                    urlObj.protocol !== "https:" &&
+                    urlObj.protocol !== "http:"
+                  ) {
+                    return `URL=${value} has invalid protocol, only allow http or https`;
+                  }
+                  // Check duplicated URL
+                  if (
+                    (dataSource.config as WebSourceConfig[]).some(
+                      (c) => c.baseUrl === value,
+                    )
+                  ) {
+                    return `URL=${value} is already added. Please provide a different URL.`;
+                  }
+                  return true;
+                },
               },
-            };
-            break;
-          case "localFolder":
-            program.engine = "context";
-            program.dataSource = {
-              type: "folder",
-              config: {
-                paths: await selectLocalContextData("folder"),
+              {
+                type: "toggle",
+                name: "shouldContinue",
+                message: "Would you like to add another website?",
+                initial: false,
+                active: "Yes",
+                inactive: "No",
               },
-            };
-            break;
-          case "web":
-            program.engine = "context";
-            program.dataSource.type = "web";
-            break;
+            ];
+            let { shouldContinue, baseUrl } = await prompts(
+              questions,
+              handlers,
+            );
+            (dataSource.config as WebSourceConfig[]).push({
+              baseUrl: baseUrl,
+              prefix: baseUrl,
+              depth: 1,
+            });
+            if (shouldContinue !== undefined && !shouldContinue) {
+              break;
+            }
+          }
+          program.dataSources.push(dataSource);
+        }
+
+        // No need to ask for another data source if user selected example data
+        if (selectedSource === "exampleFile") {
+          break;
         }
       }
+
+      if (
+        program.dataSources.length === 0 ||
+        program.dataSources[0].type === "none"
```

Comment:
can we get rid of this none type by keeping the array empty (means no datasource?)